model-optimizer/extensions/middle/StridedSliceNormalizer.py

```diff
@@ -0,0 +1,202 @@
+    after permutation to NCHW output = input[:, :, 0:50, :]. Above is show only for begin input, for end and strides
+    changes are analogously.
+
+    2nd case example
+    BEFORE:
+                  |
+                begin
+           value=[1, 50]
+                  |
+
+    AFTER:
+                  |
+                begin
+           value=[1, 1, 1]
+                  |
+             VariadicSplit
+           /              \
+          /                \
+         /       Const      \
+         \     val=[0, 0]   /
+          \       |        /
+           \      |       /
+               Concat
+           value=[1, 0, 0, 1, 1]
+                  |
+
+    Input of a shape [16, 10, 100, 100, 3] in NDHWC layout output = input[1:4, ..., 1:51, 1:3], output_shape = [3, 10, 100, 50, 2].
+    In order to do correctly layout permutation in slice expression
+    input[1:4, ..., 1:51, 1:3] ellipsis should be exended => input[1:4, :, :, 1:51, 1:3].
+    after layour permutation [1:4, 1:3, :, : 1:5]. In the places of colons blank zero begin, end and strides values
+    should be inserted. In order to do that we split begin, and concatenate with the blank zeros in the middle. Above
+    is show only for begin input, for end and strides changes are analogously.
+    """
+    enabled = True
+
+    def run_before(self):
+        from extensions.middle.LayoutChangeForConstantShapePaths import LayoutChangeForConstantShapePaths
+        return [LayoutChangeForConstantShapePaths]
+
+    def find_and_replace_pattern(self, graph: Graph):
+        ss_nodes = graph.get_op_nodes(op='StridedSlice')
+        for node in ss_nodes:
+            self.normalize_strided_slice(graph, node)
+
+            PermuteAttrs.create_permute_attrs(node, attrs=[('shrink_axis_mask', 'input:0'),
+                                                           ('new_axis_mask', 'input:0'),
+                                                           ('ellipsis_mask', 'input:0'),
+                                                           ('begin_mask', 'input:0'),
+                                                           ('end_mask', 'input:0')])
+
+            if len(node.in_port(0).data.get_shape()) < 4 and len(node.out_port(0).data.get_shape()) > 3:
+                # for the cases when we insert new_axis
+                PermuteInputs().set_input_permutation(node.in_node(1), node, 'output:0', 'shape')
+                PermuteInputs().set_input_permutation(node.in_node(2), node, 'output:0', 'shape')
+                PermuteInputs().set_input_permutation(node.in_node(3), node, 'output:0', 'shape')
+            else:
+                PermuteInputs().set_input_permutation(node.in_node(1), node, 'input:0', 'shape')
+                PermuteInputs().set_input_permutation(node.in_node(2), node, 'input:0', 'shape')
+                PermuteInputs().set_input_permutation(node.in_node(3), node, 'input:0', 'shape')
```

Comment:
Permute attributes and inputs should be set in the shape inference function.

Reply:
I was not able to successfully do that. After insertion of Concats/Split in StridedSliceNormalizer edge attribute with permutation appears on a Concat->SridedSlice edge which is not correct. And the least expensive way to avoid this is to specify permutation after normalizer.

Reply:
Added comment why permutation must be done here